### DISCUSSION TITLE
Event Bubbling for Component-Emitted Events

### DISCUSSION THREAD

**Discussion by dgtalForest** [mintysolutions/vue_snippets#21] [Q&A] [answered]
Hey all, 
I've been working with Vue components, and I noticed that component-emitted events don't bubble like native DOM events. I understand the documentation recommends using event buses or global state management to communicate between deeply nested components, but wouldn't it be more intuitive if component-emitted events could bubble up the component tree? 
It feels like a natural way to propagate events without extra setup. What do you all think? Is there a reason Vue doesn't support this?

**Comment by jdnheaps** [comment]
That's an interesting point, @dgtalForest ! I ran into the same issue recently. I needed to pass data between a deeply nested child and a parent without adding Vuex or an event bus. I ended up using $emit with multiple levels, which worked but felt a bit clunky. If events could bubble, it would definitely reduce boilerplate. But I’m wondering, would that affect performance or lead to unpredictable behavior? I guess that's something to consider.

**Comment by alm6666** [comment]
Good question, @dgtalForest ! Keaton brings up a good point about potential performance concerns. If we let component-emitted events bubble, wouldn’t it introduce complexity in managing event propagation? Right now, it's clear that parent-child communication is direct, and bubbling might create unintended side effects. Also, what if some components intercept and block events unintentionally? We'd need a way to manage that properly.

**Comment by lodistack** [comment, marked as answer]
Great discussion so far! You both raised valid concerns. In theory, adding bubbling to component events could simplify communication in some cases, but it does bring the risk of performance degradation and accidental event handling higher up the component tree, just like with native DOM events. Vue's current design intentionally avoids this complexity by keeping event communication explicit and scoped. However, it might be worth exploring an opt-in solution for bubbling, maybe with a modifier like .bubble, so developers can choose when and where they want it.

**Comment by dgtalForest** [comment]
That makes sense. An opt-in modifier sounds like a good middle ground. It keeps the default behavior clear but gives us flexibility when needed.

**Comment by jdnheaps** [comment]
Yeah, I agree. That would prevent any unexpected behavior while still giving us the option to use bubbling when it’s appropriate.

**Comment by alm6666** [comment]
Absolutely, an opt-in system sounds perfect. That way, we control which events bubble and avoid the downsides. Thanks for the insight, Ryan!
